openwisp_network_topology/integrations/device/tests.py

```diff
@@ -336,3 +335,19 @@ def test_link_change_list_queries(self):
         path = reverse('{0}_link_changelist'.format(self.prefix))
         with self.assertNumQueries(7):
             self.client.get(path)
+
+    def test_link_node_different_topology(self):
```

Comment:
The test was not good enough, because it was checking only if a ValidationError was being raised instead of checking for the specific error being raised, which can generate false positives.